connectors/http-connector/src/main/java/org/openstack4j/connectors/http/HttpCommand.java

```diff
@@ -0,0 +1,190 @@
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.List;
+import java.util.Map;
+import org.openstack4j.core.transport.HttpRequest;
+import org.openstack4j.core.transport.HttpResponse;
+import org.openstack4j.core.transport.ObjectMapperSingleton;
+import org.openstack4j.core.transport.functions.EndpointURIFromRequestFunction;
+import org.openstack4j.openstack.logging.Logger;
+import org.openstack4j.openstack.logging.LoggerFactory;
+
+/**
+ * HttpCommand is responsible for executing the actual request driven by the
+ * HttpExecutor.
+ *
+ * @param <R>
+ */
+public final class HttpCommand<R> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HttpCommand.class);
+
+    private HttpRequest<R> request;
+    private URL connectionUrl;
+    private HttpURLConnection connection;
+    private int retries;
+
+    private HttpCommand(HttpRequest<R> request) {
+
+        this.request = request;
+    }
+
+    /**
+     * Creates a new HttpCommand from the given request
+     *
+     * @param request the request
+     * @return the command
+     */
+    public static <R> HttpCommand<R> create(HttpRequest<R> request) {
+        HttpCommand<R> command = new HttpCommand<R>(request);
+        command.initialize();
+        return command;
+    }
+
+    private void initialize() {
+        try {
+
+            populateQueryParams();
+            populateHeaders();
+        } catch (Exception ex) {
+            ex.printStackTrace(System.err);
+        }
+
+    }
+
+    /**
+     * Executes the command and returns the Response
+     *
+     * @return the response
+     * @throws Exception
+     */
+    public HttpResponse execute() throws Exception {
+        StringBuilder requestBody = new StringBuilder();
+        if (request.getEntity() != null) {
+            String content = ObjectMapperSingleton.getContext(request.getEntity().getClass()).writer().writeValueAsString(request.getEntity());
+            requestBody.append(content);
+
+        } else if (request.hasJson()) {
+            requestBody.append(request.getJson());
+        }
+
+        StringBuilder contentBuilder = new StringBuilder();
+        BufferedReader in = null;
+        try {
+            connection.setRequestMethod(request.getMethod().name());
+            if (requestBody.length() > 0) {
+                System.out.println(requestBody.toString());
+                connection.setDoOutput(true);
+                BufferedOutputStream out = new BufferedOutputStream(connection.getOutputStream());
+                out.write(requestBody.toString().getBytes());
+                out.flush();
+            }
+
+            in = new BufferedReader(new InputStreamReader(
+                    connection.getInputStream()));
+            String inputLine;
+
+            while ((inputLine = in.readLine()) != null) {
+                contentBuilder.append(inputLine);
```

Comment:
I'm a little worried about this since we are taking the input stream and converting it to a String.  I have a feeling Swift (S3) will be broken streaming files back down, etc.   Could we pass the input stream to the Response object so if OS4j calls the connectors response getInputStream for binary downloads and in cases where like readyEntity is called you could apply the conversion to a String above within the response object.